domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaygroup/TabShowConditionValidatorImpl.java

```diff
@@ -0,0 +1,102 @@
+
+            if (!isInTabDisplayGroups(allTabDisplayGroups, showConditionField)) {
+                validationResult.addError(new DisplayGroupInvalidTabShowCondition(showConditionField, thisDisplayGroup));
+            }
+        }
+
+        if (tabFieldsPreconditions(thisDisplayGroup)) {
+            for (DisplayGroupCaseFieldEntity caseField : thisDisplayGroup.getDisplayGroupCaseFields()) {
+                if (caseField.getShowCondition() != null) {
+                    ShowCondition showCondition;
+                    try {
+                        showCondition = showConditionParser.parseShowCondition(caseField.getShowCondition());
+                    } catch (InvalidShowConditionException e) {
+                        validationResult.addError(new DisplayGroupInvalidTabFieldShowCondition(caseField));
+                        return validationResult;
+                    }
+
+                    String showConditionField = showCondition.getField();
+
+                    if (!isInTabDisplayGroups(allTabDisplayGroups, showConditionField)) {
+                        validationResult.addError(new DisplayGroupInvalidTabFieldShowCondition(showConditionField, caseField));
+                    }
+                }
+            }
+        }
+        return validationResult;
+    }
+
+    private List getAllTabDisplayGroups(DisplayGroupEntity thisDisplayGroup, Collection<DisplayGroupEntity> allDisplayGroups) {
+        return allDisplayGroups
+            .stream()
+            .filter(dg -> dg.getType() == DisplayGroupType.TAB)
+            .filter(dg -> dg.getCaseType().getReference().equals(thisDisplayGroup.getCaseType().getReference()))
+            .collect(Collectors.toList());
+    }
+
+    private boolean isInTabDisplayGroups(List<DisplayGroupEntity> tabDisplayGroups, String showConditionField) {
+        return tabDisplayGroups.stream().filter(tdg -> tdg.hasField(showConditionField)).findFirst().isPresent();
+    }
+
+    public boolean tabPreconditions(DisplayGroupEntity displayGroup) {
+        List<Predicate<DisplayGroupEntity>> preconditions = new ArrayList<>();
+        preconditions.add(dg -> !StringUtils.isBlank(dg.getShowCondition()));
+        preconditions.add(dg -> dg.getType() == DisplayGroupType.TAB);
+        return preconditions.stream().allMatch(p -> p.test(displayGroup));
+    }
+
+    public boolean tabFieldsPreconditions(DisplayGroupEntity displayGroup) {
+        List<Predicate<DisplayGroupEntity>> preconditions = new ArrayList<>();
+        preconditions.add(dg -> !isAllDisplayGroupCaseFieldsShowConditionBlank(dg));
+        preconditions.add(dg -> dg.getType() == DisplayGroupType.TAB);
+        return preconditions.stream().allMatch(p -> p.test(displayGroup));
+    }
+
+    private boolean isAllDisplayGroupCaseFieldsShowConditionBlank(DisplayGroupEntity dg) {
+        return dg.getDisplayGroupCaseFields().stream().allMatch(dgcf -> StringUtils.isBlank(dgcf.getShowCondition()));
```

Comment:
perhaps this is a method we can move to DisplayGroupEntity itself?

Reply:
good point